app/controllers/api/base_controller/renderer.rb

```diff
@@ -252,11 +252,11 @@ def split_virtual_attribute(attr)
       end
 
       def attr_accessible?(object, attr)
-        return false unless object && object.respond_to?(attr)
-        object.class.has_attribute?(attr) ||
-          object.class.reflect_on_association(attr) ||
-          object.class.virtual_attribute?(attr) ||
-          object.class.virtual_reflection?(attr)
+        return true if object && object.respond_to?(attr)
+        (object.class.respond_to?(:has_attribute?) && object.class.has_attribute?(attr)) ||
```

Comment:
Could you use `.try` for all of these `.respond_to?` calls? I'm not sure if it's better execution wise but it would sure be more readable. What do you think?